tests/test_perturb.py

```diff
@@ -106,43 +106,24 @@ def get_fake_ics(self, inputs, test_pt_z):
 
         return ics
 
-    def test_lowres_perturb(self, inputs_low, test_pt_z):
+    @pytest.mark.parametrize("inputs", ["inputs_low", "inputs_zel"])
+    def test_lowres_perturb(self, inputs, test_pt_z, request):
         """Tests low-resolution perturbation."""
-        ics = self.get_fake_ics(inputs_low, test_pt_z)
-        expected_dens = np.roll(ics.get("lowres_density"), (0, 1, -1), (0, 1, 2))
-        d_z_i = cf.get_growth_factor(
-            inputs=inputs_low, redshift=inputs_low.simulation_options.INITIAL_REDSHIFT
-        )
-        expected_dens *= d_z_i
-        pt = perturb_field(
-            initial_conditions=ics,
-            redshift=test_pt_z,
-            regenerate=True,
-            write=False,
-        )
-        np.testing.assert_allclose(pt.get("density"), expected_dens, atol=1e-3)
-
-    def test_zel_perturb(self, inputs_zel, test_pt_z):
-        """Tests Zeldovich perturbation."""
-        ics = self.get_fake_ics(inputs_zel, test_pt_z)
-        expected_dens = np.roll(ics.get("lowres_density"), (0, 1, 0), (0, 1, 2))
-        d_z_i = cf.get_growth_factor(
-            inputs=inputs_zel, redshift=inputs_zel.simulation_options.INITIAL_REDSHIFT
-        )
-        expected_dens *= d_z_i
-        pt = perturb_field(
-            initial_conditions=ics,
-            redshift=test_pt_z,
-            regenerate=True,
-            write=False,
-        )
-        np.testing.assert_allclose(pt.get("density"), expected_dens, atol=1e-3)
-
-    def test_linear_perturb(self, inputs_linear, test_pt_z):
-        """Tests linear perturbation."""
-        ics = self.get_fake_ics(inputs_linear, test_pt_z)
-        expected_dens = ics.get("lowres_density").copy()
-        d_z = cf.get_growth_factor(inputs=inputs_linear, redshift=test_pt_z)
+        inputs = request.getfixturevalue(inputs)
+        ics = self.get_fake_ics(inputs, test_pt_z)
+        z_d = (
+            test_pt_z
+            if inputs.matter_options.PERTURB_ALGORITHM == "LINEAR"
+            else inputs.simulation_options.INITIAL_REDSHIFT
+        )
+        roll_var = {
+            "LINEAR": (0, 0, 0),
+            "ZELDOVICH": (0, 1, 0),
+            "2LPT": (0, 1, -1),
+        }[inputs.matter_options.PERTURB_ALGORITHM]
+        d_z = cf.get_growth_factor(inputs=inputs, redshift=z_d)
+
+        expected_dens = np.roll(ics.get("lowres_density"), roll_var, (0, 1, 2))
```

Comment:
Can you add a brief comment above this... it's not obvious what the rolling is supposed to do